### ISSUE TITLE
ax.clabel returns unexpected 'NoneType'

### ISSUE THREAD

**Issue by friedrichknuth** [SciTools/cartopy#1755]
### Issue
`ax.clabel()` returns unexpected `'NoneType'`

#### Code to reproduce

Matplotlib example

```
import matplotlib
import numpy as np
import matplotlib.cm as cm
import matplotlib.pyplot as plt

delta = 0.025
x = np.arange(-3.0, 3.0, delta)
y = np.arange(-2.0, 2.0, delta)
X, Y = np.meshgrid(x, y)
Z1 = np.exp(-X**2 - Y**2)
Z2 = np.exp(-(X - 1)**2 - (Y - 1)**2)
Z = (Z1 - Z2) * 2

fig, ax = plt.subplots()
CS = ax.contour(X, Y, Z)
CS_labels = ax.clabel(CS, inline=True, fontsize=10)
print(CS_labels[0].get_position())
```
returns:
`(0.9499999999999864, 0.5360418495133943)`

Cartopy example

```
import cartopy.crs as ccrs
import matplotlib.pyplot as plt
from cartopy.examples.waves import sample_data

fig = plt.figure()

ax = fig.add_subplot(1, 1, 1, projection=ccrs.EckertIII())
ax.set_global()
ax.coastlines('110m', alpha=0.1)

x, y, z = sample_data((20, 40))
z = z * -1.5 * y

filled_c = ax.contourf(x, y, z, transform=ccrs.PlateCarree())
line_c = ax.contour(x, y, z, levels=filled_c.levels,
                    colors=['black'],
                    transform=ccrs.PlateCarree())

CS_labels = ax.clabel(
    line_c, 
    colors=['black'],
    manual=False,
    inline=True, 
    fmt=' {:.0f} '.format,
)

print(CS_labels[0].get_position())
```
breaks with the following:
#### Traceback 

```
TypeError                                 Traceback (most recent call last)
<ipython-input-655-34edf4c96e8c> in <module>
----> 1 CS_labels[0].get_position()

TypeError: 'NoneType' object is not subscriptable
```

#### Additional context

Related question and solution to accessing label coordinates on [stackoverflow](https://stackoverflow.com/questions/66807997/get-matplotlib-cartopy-contour-auto-label-coordinates).

**Comment by greglucas** [comment]
I believe this is the same as #1660 and should be fixed in the next release. Can you try with the release candidate for 0.19?
`pip install cartopy==0.19.0rc1`

**Comment by greglucas** [comment]
This was fixed with the 0.19 release.
